salt/states/idem.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+#
+# Author: Tyler Johnson <tjohnson@saltstack.com>
+#
+
+"""
+Idem Support
+============
+
+This state provides access to idem states
+"""
+from __future__ import absolute_import, print_function, unicode_literals
+
+import logging
+import os
+import sys
+
+try:
+    import pop.hub
+
+    HAS_LIBS = True, None
+except ImportError as e:
+    HAS_LIBS = False, str(e)
+
+log = logging.getLogger(__name__)
+
+# Function alias to make sure not to shadow built-in's
+__func_alias__ = {"exec_": "exec"}
+__virtualname__ = "idem"
+
+
+def __virtual__():
+    if sys.version_info < (3, 6):
+        return False, "idem only works on python3.6 and later"
+    if not HAS_LIBS[0]:
+        return HAS_LIBS
+    return __virtualname__
+
+
+def __init__(opts):
+    if "idem.hub" not in __context__:
+        # Initialize the hub
+        log.debug("Creating the POP hub")
+        hub = pop.hub.Hub()
+        log.debug("Initializing the loop")
+        hub.pop.loop.create()
+        # Load idem grains/states/exec modules onto the hub
+        log.debug("Loading subs onto hub")
+        hub.pop.sub.add(dyne_name="acct")
+        hub.pop.sub.add(dyne_name="config")
+        hub.pop.sub.add(dyne_name="idem")
+        hub.pop.sub.add(dyne_name="exec")
+        hub.pop.sub.add(dyne_name="states")
+        log.debug("Reading idem config options")
+        hub.config.integrate.load(["acct", "idem"], "idem", parse_cli=False, logs=False)
+        __context__["idem.hub"] = hub
+
+
+def _get_refs(hub, sources):
```

Comment:
In case it's desired, this doesn't support salt environments

Reply:
Idem does not support salt environments.  Instead it uses `--sls-sources`, `--tree`, and `acct`.  I added support for acct, but maybe I need to add a way to set sls-sources and tree

Reply:
Ah, I wasn't sure if you intended to support salt:// urls for file serving. Integration with salt could mean that idem sls files live on the master (like other salt states), and are transparently copied to the minion for idem to render/run. But this integration is aiming for only aiming to pass local file paths to idem. 👍

Reply:
For now idem states are only intended to be run on the master minion